docs/book/checkout.rst

```diff
@@ -0,0 +1,259 @@
+
+.. code-block:: yaml
+
+   transitions:
+      address:
+          from: [cart]
+          to: addressed
+      readdress:
+          from: [payment_selected, shipping_selected, addressed]
+          to: cart
+      select_shipping:
+          from: [addressed]
+          to: shipping_selected
+      reselect_shipping:
+          from: [payment_selected, shipping_selected]
+          to: addressed
+      select_payment:
+          from: [shipping_selected]
+          to: payment_selected
+      reselect_payment:
+          from: [payment_selected]
+          to: shipping_selected
+      complete:
+          from: [payment_selected]
+          to: completed
+
+Steps of Checkout
+-----------------
+
+Checkout in Sylius is divided into 4 steps. Each of these steps occurs when the Order goes into a certain state.
+See the Checkout state machine in the ``CoreBundle\Resources\config\app\state_machine.yml``
+together with the routing file for checkout: ``ShopBundle\Resources\config\routing\checkout.yml``.
+
+.. note::
+
+    Before performing Checkout :doc:`you need to have an Order created </book/orders>`.
+
+Addressing
+~~~~~~~~~~
+
+This is a step where the customer provides both **shipping and billing addresses**.
+
++--------------------------+----------------------------------------------------+
+| Transition after step    | Template                                           |
++--------------------------+----------------------------------------------------+
+| ``cart``-> ``addressed`` | ``SyliusShopBundle:Checkout:addressing.html.twig`` |
++--------------------------+----------------------------------------------------+
+
+How to perform the Addressing Step programmatically?
+''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+Firstly if the **Customer** is not yet set on the Order it will be assigned depending on the case:
+
+* An already logged in **User** - the Customer is set for the Order using the ``Sylius\Bundle\CoreBundle\EventListener\CartBlamerListener``, that determines the user basing on the event.
+* An existent **User** that is not logged in - they are redirected to log in before continuing.
+* A **Customer** that was present in the system before (we've got their e-mail) - the Customer instance is updated via cascade, the order is assigned to it.
+* A new **Customer** with unknown e-mail - a new Customer instance is created and assigned to the order.
+
+Of course the customer data like name and surname is also handled.
+
+The typical **Address** consists of: country, city, street and postcode - to assign it to an Order either create it manually or retrieve from the repository.
+
+.. code-block:: php
+
+     /** @var AddressInterface $address */
+     $address = $this->container->get('sylius.factory.address')->createNew();
+
+     $address->setFirstName('Name');
+     $address->setLastName('Surname');
+     $address->setStreet('Street');
+     $address->setCountryCode('PL');
```

Comment:
Why rest of the samples have no meaning, and here we have direct value? Maybe it would be worthy to put sample data to all of fields. I don't know, an address of white house or something similar?